Dockerfile

```diff
@@ -1,23 +1,33 @@
-FROM python:3.10-slim
-
-LABEL name="httpbin"
-LABEL version="0.9.2"
-LABEL description="A simple HTTP service."
-LABEL org.kennethreitz.vendor="Kenneth Reitz"
+FROM python:3.10-slim AS build
 
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apt update -y && apt install python3-pip git -y && pip3 install --no-cache-dir pipenv
+RUN apt-get -y update
+RUN apt-get install -y \
+    python3-pip \
+    python3-venv
 
-ADD Pipfile Pipfile.lock /httpbin/
-WORKDIR /httpbin
-RUN /bin/bash -c "pip3 install --no-cache-dir -r <(pipenv requirements)"
+RUN python3 -m venv /opt/httpbin
+RUN /opt/httpbin/bin/pip install -U pip
 
 ADD . /httpbin
-RUN pip3 install --no-cache-dir /httpbin
-RUN chmod +x /httpbin/httpbin.bash
+RUN /opt/httpbin/bin/pip install /httpbin[mainapp]
 
-EXPOSE 80
 
-CMD ["/httpbin/httpbin.bash"]
+# ----------------------------------------------------------------------------
+
+FROM python:3.10-slim AS prod
+
+ARG APP_VERSION
+LABEL name="httpbin"
+LABEL version=${APP_VERSION}
+LABEL description="A simple HTTP service."
+LABEL org.kennethreitz.vendor="Kenneth Reitz"
+
+COPY --from=build /opt/httpbin /opt/httpbin
+ADD httpbin.bash /opt/httpbin/bin
+RUN chmod +x /opt/httpbin/bin/httpbin.bash
+EXPOSE 80
```

Comment:
We're breaking the API of where to get the image from, right? If we're doing that, why not also not use root and expose a different port. People can always remap them on the outside as they wish.

Would include adding a `USER httpbin` and running the service in the bash script with a different port but that should be fine.

Reply:
I was thinking the same thing but wanted to keep the external changes (of the running image) to a minimum.

If you are okay with that change I will gladly do it as it will simplify deployments in non-privileged docker runtimes (like OpenShift) which enforce non-root images.

Reply:
I have pushed the necessary changes in 4923332

If we decide to keep the old external behaviour (port 80 by default) we can easily revert that commit. With this change, the `HTTPBIN_PORT` customisation that I provided in #4 becomes moot and we could drop that again. It messes with the `EXPOSE` information of the image anyway and was only added as a workaround to make the image run on environments that enforce non-root containers.

Should I remove that again? Only one other user expressed interest and it was only released for a short time. I'm generally against dropping something once it's released. But maybe we could make an exception here?